plugins/hls-eval-plugin/src/Ide/Plugin/Eval/Code.hs

```diff
@@ -99,12 +97,19 @@ evalSetup = do
     preludeAsP <- parseImportDecl "import qualified Prelude as P"
     context <- getContext
     setContext (IIDecl preludeAsP : context)
-    mapM_
-        runDecls
-        [ "class Print f where asPrint :: f -> P.IO P.String"
-        , "instance P.Show a => Print (P.IO a) where asPrint io = io P.>>= P.return P.. P.show"
-        , "instance P.Show a => Print a where asPrint a = P.return (P.show a)"
-        ]
+
+-- | A wrapper of 'InteractiveEval.execStmt', capturing the execution result
+myExecStmt :: String -> ExecOptions -> Ghc (Either String (Maybe String))
+myExecStmt stmt opts = do
+    (temp, purge) <- liftIO newTempFile
+    evalPrint <- head <$> runDecls ("evalPrint x = P.writeFile \"" <> temp <> "\" (P.show x)")
+    modifySession $ \hsc -> hsc {hsc_IC = setInteractivePrintName (hsc_IC hsc) evalPrint}
+    result <- execStmt stmt opts >>= \case
+              ExecComplete (Left err) _ -> pure $ Left $ show err
+              ExecComplete (Right _) _ -> liftIO $ Right . (\x -> if null x then Nothing else Just x ) <$> readFile temp
```

Comment:
use the strict [`readFile'`](https://hackage.haskell.org/package/extra-1.7.9/docs/Extra.html#v:readFile-39-)